src/java.base/share/native/libcrexec/image_score.cpp

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2025, Azul Systems, Inc. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+#include <cerrno>
+#include <cstdio>
+#include <cstring>
+
+#include "crexec.hpp"
+#include "image_score.hpp"
+
+bool ImageScore::persist(const char* image_location) const {
+  char fname[PATH_MAX];
+  if (snprintf(fname, sizeof(fname), "%s/score", image_location) >= (int) sizeof(fname) - 1) {
+    fprintf(stderr, CREXEC "filename too long: %s/score\n", image_location);
+    return false;
+  }
+  FILE* f = fopen(fname, "w");
+  if (f == nullptr) {
+    fprintf(stderr, CREXEC "cannot open %s for writing: %s\n", fname, strerror(errno));
+    return false;
+  }
+  _score.foreach([&](const Score& score){
+    fprintf(f, "%s=%f\n", score._name, score._value);
```

Comment:
There's no validation that `=` and `\n` are not present in the name so there will probably be problems with parsing